Source/WebDriver/WebDriverService.cpp

```diff
@@ -2732,32 +2733,79 @@ void WebDriverService::bidiSessionSubscribe(unsigned id, RefPtr<JSON::Object>&&p
     // https://bugs.webkit.org/show_bug.cgi?id=282436
     // FIXME: Support by-context subscriptions.
     // https://bugs.webkit.org/show_bug.cgi?id=282981
-    for (auto& eventName : *eventNames) {
-        auto event = eventName->asString();
-        m_session->enableGlobalEvent(event);
+    Vector<String> eventNames;
+    for (auto& eventNameJS : *eventNamesJS) {
+        auto eventName = eventNameJS->asString();
+        if (!eventName) {
+            completionHandler(WebSocketMessageHandler::Message::fail(CommandResult::ErrorCode::InvalidArgument, std::nullopt, "Invalid event name"_s, id));
+            return;
+        }
+        eventNames.append(eventName);
     }
+    m_session->subscribe(WTFMove(eventNames), { }, { }, [id, completionHandler = WTFMove(completionHandler)](CommandResult&& result) mutable {
+        if (result.isError()) {
+            auto errorMessage = result.errorMessage() ? result.errorMessage() : "Failed to subscribe"_s;
+            completionHandler(WebSocketMessageHandler::Message::fail(CommandResult::ErrorCode::InvalidArgument, std::nullopt, errorMessage, id));
+            return;
+        }
+        auto subscriptionObj = JSON::Object::create();
+        subscriptionObj->setString("subscription"_s, result.result()->asString());
 
-    completionHandler(WebSocketMessageHandler::Message::reply("success"_s, id, JSON::Value::null()));
+        completionHandler(WebSocketMessageHandler::Message::reply("success"_s, id, WTFMove(subscriptionObj)));
```

Comment:
Drive-by feedback: it's a little weird that the first argument to `WebSocketMessageHandler::Message::reply` is a string rather than an enum. Is there a reason to do it this way?

I think it's implied that it's a success, since it's calling `reply()` and not `fail()` right?